### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by michalmuskala**

### CODE REVIEW

File: lib/ex_aws/cognito_idp.ex [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+  # TODO: update_user_pool_client
+  # TODO: verify_user_attribute
+
+  defp request(action, data) do
+    headers = [
+      {"x-amz-target", "#{@namespace}.#{action}"},
+      {"content-type", "application/x-amz-json-1.1"}
+    ]
+
+    ExAws.Operation.JSON.new(:"cognito-idp", data: data, headers: headers)
```
Comment (michalmuskala) [inline lib/ex_aws/cognito_idp.ex:396]: @benwilson512 I tried using `:cognito_idp` as you suggested, but it failed with signature errors:
```elixir
%{
  "__type" => "InvalidSignatureException", 
  "message" => "Credential should be scoped to correct service: 'cognito-idp'. "
}
```

File: mix.exs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+defmodule ExAws.Cognito.MixProject do
```
Comment (benwilson512) [inline mix.exs]: This probably should be `ExAws.CognitoIdp.MixProject` right?

File: mix.exs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  use Mix.Project
+
+  def project do
+    [
+      app: :ex_aws_cognito,
```
Comment (benwilson512) [inline mix.exs]: This should probably be `:ex_aws_cognito_idp`?

File: lib/ex_aws/cognito_idp.ex (hunk 1 above)
Comment (benwilson512) [reply lib/ex_aws/cognito_idp.ex:396]: I'm thinking we may want to just leave this alone as `:"cognito-idp"`. While slightly less than ergonomic it's rare to actually need to type it, and I'm concerned that it's going to be difficult to try to normalize this at a central place.

**Review by benwilson512 — CHANGES_REQUESTED** [review]

File: lib/ex_aws/cognito_idp.ex (hunk 1 above)
Comment (michalmuskala) [reply lib/ex_aws/cognito_idp.ex:396]: Yeah, it's just in 3 places, I'm completely fine with leaving it as `:"cognito-idp"`

**Comment by michalmuskala** [comment]
@benwilson512 updated

**Comment by benwilson512** [comment]
Hi @michalmuskala. If you wish to be a maintainer of `ex_aws_cognito_idp` I'd be happy to add you, and then you can merge this or any other branch.

**Comment by michalmuskala** [comment]
Hi Ben. I was primarily interested in this since I used it at the time at a company I worked for. I moved since then and I'm not really interested in maintaining this myself.

**Pull request closed without merging**